--- a/portability/meson.build
+++ b/portability/meson.build
@@ -3,9 +3,7 @@ cc = meson.get_compiler('c')
 libportability_src = []
 
 check_symbols = [
-	['getrandom', 'getrandom.c', 'NEED_GETRANDOM', 'sys/random.h'],
 	['memrchr', 'memrchr.c', 'NEED_MEMRCHR', 'string.h'],
-	['memfd_create', '', 'NEED_MEMFD_CREATE', 'sys/mman.h'],
 	['mknodat', 'mknodat.c', 'NEED_MKNODAT', 'sys/stat.h'],
 	['pipe2', 'pipe2.c', 'NEED_PIPE2', 'unistd.h'],
 	['qsort_r', 'qsort_r.c', 'NEED_QSORT_R', 'stdlib.h'],
--- a/src/package.c
+++ b/src/package.c
@@ -756,15 +756,15 @@ int apk_ipkg_run_script(struct apk_insta
 
 	argv[0] = fn;
 
-	if (!db->memfd_failed) {
-		/* Linux kernel >= 6.3 */
-		fd = memfd_create(fn, MFD_EXEC);
-		if (fd < 0 && errno == EINVAL) {
-			/* Linux kernel < 6.3 */
-			fd = memfd_create(fn, 0);
-			if (fd < 0) db->memfd_failed = 1;
-		}
-	}
+//	if (!db->memfd_failed) {
+//		/* Linux kernel >= 6.3 */
+//		fd = memfd_create(fn, MFD_EXEC);
+//		if (fd < 0 && errno == EINVAL) {
+//			/* Linux kernel < 6.3 */
+//			fd = memfd_create(fn, 0);
+//			if (fd < 0) db->memfd_failed = 1;
+//		}
+//	}
 	if (!db->script_dirs_checked) {
 		if (fd < 0 && apk_make_dirs(root_fd, script_exec_dir, 0700, 0755) < 0) {
 			reason = "failed to prepare dirs for hook scripts: ";
